### PR TITLE
Print short fault info only from safe fields

### PR DESCRIPTION
**Pull request by davenger**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
We recently encountered double segfaults: a segfault while printing info on the previous segfault. In that case we didn't manage to log any info on the first fault. With this change we will try to log some safe info first before accessing potentially unsafe thead_ptr

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
This is an automated comment for commit 602392bb6206590e0d24df05eabf69a970767756 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51932/602392bb6206590e0d24df05eabf69a970767756/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Daemon/BaseDaemon.cpp [hunk 1]
```diff
@@ -336,30 +324,22 @@ class SignalListener : public Poco::Runnable
         else if (sig >= 0)
             signal_description = strsignal(sig); // NOLINT(concurrency-mt-unsafe) // it is not thread-safe but ok in this context
```
Comment (tavplubix) [inline src/Daemon/BaseDaemon.cpp:325]: Let's also replace it with `sigdescr_np` just in case

**Review by tavplubix — APPROVED** [review]

File: src/Daemon/BaseDaemon.cpp (hunk 1 above)
Comment (davenger) [reply src/Daemon/BaseDaemon.cpp:331]: Looks like we don't have `sigdescr_np` in our contirbs.
I've rearranged the code to print signal number first.

**Comment by alexey-milovidov** [comment]
Also needed for https://s3.amazonaws.com/clickhouse-test-reports/0/fa9514f21612e60c0939b7bfd1b5f4f1f1f410c8/stateless_tests__debug__[2_5].html

**Pull request merged**